src/io/iohk/decco/MD5.scala

```diff
@@ -34,7 +35,10 @@ object MD5 {
 
     override def decode(start: Int, source: ByteBuffer): Either[Failure, DecodeResult[MD5]] = {
       val dst = new Array[Byte](16)
-      if (source.remaining() < 16) {
+      val p = (source: Buffer).position()
```

Comment:
Why is forcing the types required?

Reply:
Due to a problem with JDK9. I'm not entirely sure of the details, but, without the cast, it blows up with a MethodNotFoundError on JDK9

Reply:
Are we supposed to support JDK9? why not JDK11 instead?

Reply:
a) I'm not sure if it's a problem 'with JDK9' or 'since JDK9'
b) whatever it is, it's hitting circleci. And without this, circleci fails